ndsl/debug/debugger.py

```diff
@@ -0,0 +1,109 @@
+import dataclasses
+import numbers
+import os
+import pathlib
+
+import pandas as pd
+import xarray as xr
+
+from ndsl.logging import ndsl_log
+from ndsl.quantity import Quantity
+
+
+@dataclasses.dataclass
+class Debugger:
+    """Debugger relying on `ndsl.debug.config` for setup capable
+    of doing automatic data save on external configuration."""
+
+    # Configuration
+    stencils_or_class: list[str] = dataclasses.field(default_factory=list)
+    track_parameter_by_name: list[str] = dataclasses.field(default_factory=list)
+    save_compute_domain_only: bool = False
+    dir_name: str = "./"
+
+    # Runtime data
+    rank: int = -1
+    calls_count: dict[str, int] = dataclasses.field(default_factory=dict)
+    track_parameter_count: dict[str, int] = dataclasses.field(default_factory=dict)
+
+    def _to_xarray(self, data, name) -> xr.DataArray:
+        if isinstance(data, Quantity):
+            if self.save_compute_domain_only:
+                mem = data.view[:]
+                shp = data.view[:].shape
```

Comment:
Not gonna use Field here?

Reply:
Good catch this was written prior, will fix that.